core/scheduler/scheduler.go

```diff
@@ -139,6 +139,12 @@ func (s *Scheduler) Run() error {
 
 // GetDutyDefinition returns the definition for a duty if resolved already, otherwise an error.
 func (s *Scheduler) GetDutyDefinition(ctx context.Context, duty core.Duty) (core.DutyDefinitionSet, error) {
+	if duty.Type == core.DutyBuilderProposer && !s.builderAPI {
+		return nil, errors.New("builder-api not enabled, but duty builder proposer requested")
+	} else if duty.Type == core.DutyProposer && s.builderAPI {
```

Comment:
does this mean if builder api is enabled then we won't use DutyProposer at all?

Reply:
yup